### PR TITLE
Make the Rust docs display properly in Netsurf browser

### PR DESCRIPTION
**Pull request by grego**
Adds minor tweaks to CSS which do not change the appearance in the major browsers.

### CODE REVIEW

File: src/librustdoc/html/static/css/themes/light.css [hunk 1]
```diff
@@ -239,7 +249,7 @@ pre.rust .question-mark {
 }
 
 a.test-arrow {
-	background-color: rgb(78, 139, 202, 0.2);
+	background-color: rgba(78, 139, 202, 0.2);
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/css/themes/light.css:252]: Nice catch.

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/themes/light.css:252]: Can you send a PR for this fix in any case please?

**Comment by GuillaumeGomez** [comment]
But if you remove it from `ayu.css` (and `light.css` too) and move it into `rustdoc.css`, it would work. But adding more rules to counter-balance bad CSS support is a bit weird. For extra explanations: `light.css` is the minimum that all other themes **must** implement. `ayu` just provides more "precise" things.

**Comment by grego** [comment]
I understand. But `ayu` sets `h1.fqn a` to `#fff`, which is not its `--main-color` (`#c5c5c5`), similarly for `.docblock code` and others, so this would break it.

**Comment by grego** [comment]
Given this, I restored the changes in the light theme.

**Comment by grego** [comment]
I did :) https://github.com/grego/rust/commit/6c6f30f45ecb894539f32f6b49e24ededb386dbf
It probably doesn't show up here because the PR is closed.

**Comment by GuillaumeGomez** [comment]
Weird github behaviour... I'll check your code locally tomorrow to see if nothing is missing.

@jsha: Do you think that the changes are acceptable with the current code?

**Comment by jsha** [comment]
Sorry, I still don't think it's a good idea to make exceptions to the browser support policy. If each of these changes was justifiable independently on a correctness basis, that would be one thing, but I think some of them are actually incorrect. For instance, `pre.rust` is set to be `inline-block`, but it shouldn't be - we never want code blocks to line-wrap with other content. I'm guessing that happens to work out because code blocks are also `width: 100%`, but that's an awkward hack that is likely to surprise someone down the line or break in a surprising way on some supported browser.

It's great that netsurf is working on HTML5 and CSS3. From the direction of this PR, I'm guessing one of the major things currently lacking is flexbox support. Once it's implemented, I suspect netsurf will display rustdoc pages nicely.

For locally generated documentation, you can use the `--html-in-header` flag to incorporate your own CSS that adds netsurf-related modifications as needed.

**Comment by GuillaumeGomez** [comment]
I was wondering if some of the changes were breaking the docs and it seems to be the case.

Still: can you send a PR for the `rgba` fix in any case please?

**Pull request closed without merging**